up-l3/udiscovery/v3/client.adoc

```diff
@@ -39,12 +42,88 @@ In <<client-api>> above, _Client_ represents the client-facing APIs to talk to u
 
 == Client Facing APIs
 
-The client facing APIs of uDiscovery provide the means for uEs to discover services (where they are located, versions, etc...), and topic metadata that a service produces (method id and permissions, publish topic message formats, etc...). The API is declared in the the xref:../../../up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto[uDiscovery service proto].
+The client facing APIs of uDiscovery provide the means for uEs to discover services (where they are located, versions, etc...), and topic metadata that a service produces (method id and permissions, publish topic message formats, etc...). The API is declared in the the {discovery-proto-ref}.
 
 [.specitem,oft-sid="dsn~discovery-client-apis~1",oft-needs="impl"]
 --
-* link:../languages.adoc[uProtocol Language Library] *MUST* implement the client-side APIs of xref:../../../up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto[udiscovery.proto] using xref:../../../up-l2/api.adoc[Communication Layer (uP-L1) RpcClient API]. 
+Each link:../languages.adoc[uProtocol Language Library] *MUST* implement the client-side APIs of {discovery-proto-ref} using the {up-l2-ref} RpcClient. 
+--
+
+=== FindServices()
+
+The FindServices() API is used to find the authority(ies), instance(s), and version(s) of services based on the passed URI. The only mandatory portion of the URI that must be passed is the ue_id, the rest can be discovered. When you pass the wildcard for authority_name, the search will be done up to the domain, in order to search in the central discovery database, you must set search_central=true.
+
+[.specitem,oft-sid="dsn~discovery-findservices-error-notfound~1",oft-needs="impl,test"]
+--
+* *MUST* return a `UCode.NOT_FOUND` if no service matching the passed URI was found. 
+--
+
+[.specitem,oft-sid="dsn~discovery-findservices-error-invalid-argument~1",oft-needs="impl,test"]
 --
+* *MUST* return a `UCode.INVALID_ARGUMENT` if: 
+  * The lower 16 bits of `ue_id` (_service ID_) is missing.
```

Comment:
how can the lower 16 bits be missing, i.e. how would a corresponding URI look like? Or did you mean: if the lower 16 bits are 0x0000?

Reply:
hum... I guess you might want to find out version for the uSubscription service so this requirement needs to be removed then all together.

Reply:
Can you add a corresponding entry to the list of examples? I still haven't understood the use case ...

Reply:
Pushing a change now